pandas_gbq/gbq.py

```diff
@@ -188,78 +204,130 @@ def get_application_default_credentials(self):
+        credentials from a file or by going through the OAuth flow.
 
-        if credentials is None or credentials.invalid or self.reauth:
-            credentials = run_flow(flow, storage, argparser.parse_args())
+        Parameters
+        ----------
+        None
 
-        return credentials
+        Returns
+        -------
+        GoogleCredentials : credentials
+            Credentials for the user with BigQuery access.
+        """
+        from google_auth_oauthlib.flow import InstalledAppFlow
+        from oauthlib.oauth2.rfc6749.errors import OAuth2Error
+
+        credentials = self.load_user_account_credentials()
+
+        client_config = {
+            'installed': {
+                'client_id': ('495642085510-k0tmvj2m941jhre2nbqka17vqpjfddtd'
+                              '.apps.googleusercontent.com'),
+                'client_secret': 'kOc9wMptUtxkcIFbtZCcrEAc',
+                'redirect_uris': ['urn:ietf:wg:oauth:2.0:oob'],
+                'auth_uri': 'https://accounts.google.com/o/oauth2/auth',
+                'token_uri': 'https://accounts.google.com/o/oauth2/token',
+            }
+        }
 
-    def get_service_account_credentials(self):
-        # Bug fix for https://github.com/pandas-dev/pandas/issues/12572
-        # We need to know that a supported version of oauth2client is installed
-        # Test that either of the following is installed:
-        # - SignedJwtAssertionCredentials from oauth2client.client
-        # - ServiceAccountCredentials from oauth2client.service_account
-        # SignedJwtAssertionCredentials is available in oauthclient < 2.0.0
-        # ServiceAccountCredentials is available in oauthclient >= 2.0.0
-        oauth2client_v1 = True
-        oauth2client_v2 = True
+        if credentials is None or self.reauth:
+            app_flow = InstalledAppFlow.from_client_config(
+                client_config, scopes=[self.scope])
 
-        try:
-            from oauth2client.client import SignedJwtAssertionCredentials
-        except ImportError:
-            oauth2client_v1 = False
+            try:
+                if self.auth_local_webserver:
+                    credentials = app_flow.run_local_server()
+                else:
+                    credentials = app_flow.run_console()
+            except OAuth2Error as ex:
+                raise AccessDenied(
+                    "Unable to get valid credentials: {0}".format(ex))
 
-        try:
-            from oauth2client.service_account import ServiceAccountCredentials
-        except ImportError:
-            oauth2client_v2 = False
+            self.save_user_account_credentials(credentials)
 
-        if not oauth2client_v1 and not oauth2client_v2:
-            raise ImportError("Missing oauth2client required for BigQuery "
-                              "service account support")
+        return credentials
 
+    def get_service_account_credentials(self):
+        import httplib2
```

Comment:
are any of these new functions user API visible? if so, do a _try_imports at the top of them.